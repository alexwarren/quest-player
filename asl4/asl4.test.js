'use strict';

const output = [];

window.quest = window.quest || {};
window.quest.print = (text) => {
    output.push(text);
};
window.quest.ui = {
    showMenu: () => {},
    beginWait: () => {},
    showQuestion: () => {},
    playSound: () => {},
    stopSound: () => {},
    clearScreen: () => {},
    setBackground: () => {},
    setPanelContents: () => {},
    panesVisible: () => {},
    locationUpdated: () => {},
    setGameName: () => {},
    show: () => {},
    updateStatus: () => {},
    updateList: () => {},
    updateCompass: () => {},
    requestNextTimerTick: () => {}
};

console.log = () => {};

const asl4 = require('./asl4');
const fs = require('fs');

const fileFetcher = function (filename, onSuccess, onFailure) {
    try {
        const data = fs.readFileSync(filename, 'utf-8');
        onSuccess(data);
    }
    catch (e) {
        onFailure();
    }
};

test('loads test.asl', async (done) => {
    const game = asl4.createGame('examples/test.asl', null, null, fileFetcher, null, null);
    const onSuccess = async () => {
        await game.Begin();
        game.SendCommand('x book');
        setTimeout(() => {
            game.SetMenuResponse('1');
            setTimeout(() => {
                expect(output).toMatchSnapshot();
                done();
            }, 1);
        }, 1);
    };
    const onFailure = () => {};
    await game.Initialise(onSuccess, onFailure);
});