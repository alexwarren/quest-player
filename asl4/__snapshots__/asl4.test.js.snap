// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads test.asl 1`] = `
Array [
  "<span style=\\"font-family:Arial;font-size:9pt\\">You are in </span><span style=\\"font-family:Arial;font-size:9pt\\"><span style=\\"color:red\\">room</span></span><span style=\\"font-family:Arial;font-size:9pt\\">.<br />There is </span><span style=\\"font-family:Arial;font-size:9pt\\"><b>thing</b></span><span style=\\"font-family:Arial;font-size:9pt\\">, a red </span><span style=\\"font-family:Arial;font-size:9pt\\"><b>book</b></span><span style=\\"font-family:Arial;font-size:9pt\\"> and a blue </span><span style=\\"font-family:Arial;font-size:9pt\\"><b>book</b></span><span style=\\"font-family:Arial;font-size:9pt\\"> here.</span>",
  "<span style=\\"font-family:Arial;font-size:9pt\\">You can go </span><span style=\\"font-family:Arial;font-size:9pt\\"><b>east</b></span><span style=\\"font-family:Arial;font-size:9pt\\">.</span>",
  "",
]
`;
